Update blog instructions to include Blog Post Starter Pack (#4186)
sourcegraph/handbook: content/departments/marketing/blog.md

## Adding a blog post

Once your draft is approved, you're ready to add a blog post by creating a Markdown file in one of the `sourcegraph/about` repository `/content/blogposts` child directories.

## Convert to markdown

To quickly convert your document into Markdown, you can use [Docs to Markdown](https://workspace.google.com/u/0/marketplace/app/docs_to_markdown/700168918607). This is not required, but can save some time if you do not prefer to rewrite the content.

> If you do this, please add the **frontmatter** to the beginning of the file, frontmatter instructions can be found in the [Blog Post Starter Pack](/departments/marketing/blog/#using-the-blog-post-starter-pack).

Reviewer: The GitHub Action is complaining about this line. I'm thinking this may be resolved by just referencing the anchor link:

```suggestion
If you do this, please add the **frontmatter** to the beginning of the file, frontmatter instructions can be found in the [Blog Post Starter Pack](#using-the-blog-post-starter-pack).
```

Reviewer 2: Ah I thought it was failing b/c the link isn't valid in prod yet. Ok this step looks good now, thanks ! 😄

## Using the Blog Post Starter Pack

The Content Platform Team maintains a template Markdown file that exposes all of the content components available for use in the blog. You can directly copy and paste components from this file into your post. It's important to use the components provided for tables, images, alerts, youtube videos, external or self hosted videos, and HubSpot forms to ensure our blog posts are uniform in styling.

Access the template in the About repository: [`/content/blogposts/blog-post-starter-pack.md`](https://github.com/sourcegraph/about/tree/main/content/blogposts/blog-post-starter-pack.md).

> To use the template, copy the above file and paste it in the appropriate year folder (i.e. `/content/blogposts/2022/`). Rename the file to the slug the post will be, or something similar. Edit the markdown content accordingly, all available standardized components are ready to be changed right in this file!

Reviewer: ```suggestion
To use the template, copy the above file and paste it in the appropriate year folder (i.e. `/content/blogposts/2022/`). Rename the file to the slug the post will be, or something similar. Edit the markdown content accordingly. All available standardized components are ready to be changed right in this file!
```

## Adding a blog post

Once your draft is approved, you're ready to add a blog post by creating a Markdown file in one of the `sourcegraph/about` repository `content/blogposts` child directories.

## Convert to markdown

To quickly convert your document into Markdown, you can use [Docs to Markdown](https://workspace.google.com/u/0/marketplace/app/docs_to_markdown/700168918607). This is not required, but can save some time if you do not prefer to rewrite the content.

> If you do this, please add the following frontmatter to the beginning of the file:

Reviewer: I would update this and move the frontmatter to the blog starter pack and use that as the source of truth for all formatting

- The `socialImage` field is optional. Use the full path to image in order to be read properly on Twitter and Facebook. Ideal image size: 1,200 x 628 pixels. <a href="https://sproutsocial.com/insights/social-media-image-sizes-guide/" rel="nofollow" target="_blank">Latest social size guidelines</a>.
- The `videoID` field is an optional YouTube video ID and will take priority even if the `socialImage` is present. This will generate an inline video preview card when sharing on social media. This is supported for all types of posts; blog, podcast, and release posts.

## Using the Blog Post Starter Pack

> The Content Platform Team maintains a template Markdown file that exposes all of the content components available for use in the blog. You can directly copy and paste components from this file into your post.

Reviewer 2: ```suggestion
The Content Platform Team maintains a template Markdown file that exposes all of the content components available for use in the blog. You can directly copy and paste components from this file into your post. It's important to use the components provided for tables, images, alerts, youtube videos, videos, and hubspot forms to ensure our blog posts are uniform in styling.
```

Some added context

> Access the template in the About repository: `/content/blogposts/blog-post-starter-pack.md`

Reviewer 2: I know this file isn't posted yet in the About repo, but it'd be helpful to have a link to the file to be really explicit here

Author: Agreed! I can add it once we merge the About PR.

## Adding images

Please use a `Figure` component to add images to your post. Optionally, you can include a `caption` to display a description below the image.

<!-- prettier-ignore -->
```html
<Figure 
  src="https://storage.googleapis.com/sourcegraph-assets/blog/default_hero_social.png"
  alt="Sourcegraph thumbnail"
  caption="The Sourcegraph logo"
/>
```

> Please do not add images using `img` tags or `![text](image.png)` methods.

Reviewer 2: This is great! Can you add a section for the other missing components here? (`Alert`, `BlockquoteWithBorder`, `TableWrapper`)

Author: Sure thing! I have included descriptions for these three here: 9d453a5

- The `socialImage` field is optional. Use the full path to image in order to be read properly on Twitter and Facebook. Ideal image size: 1,200 x 628 pixels. <a href="https://sproutsocial.com/insights/social-media-image-sizes-guide/" rel="nofollow" target="_blank">Latest social size guidelines</a>.
- The `videoID` field is an optional YouTube video ID and will take priority even if the `socialImage` is present. This will generate an inline video preview card when sharing on social media. This is supported for all types of posts; blog, podcast, and release posts.

## Using the Blog Post Starter Pack

> The Content Platform Team maintains a template Markdown file that exposes all of the content components available for use in the blog. You can directly copy and paste components from this file into your post. It's important to use the components provided for tables, images, alerts, youtube videos, videos, and hubspot forms to ensure our blog posts are uniform in styling.

Reviewer: ```suggestion
The Content Platform Team maintains a template Markdown file that exposes all of the content components available for use in the blog. You can directly copy and paste components from this file into your post. It's important to use the components provided for tables, images, alerts, youtube videos, external or self hosted videos, and HubSpot forms to ensure our blog posts are uniform in styling.
```

Access the template in the About repository: [`/content/blogposts/blog-post-starter-pack.md`](https://github.com/sourcegraph/about/tree/main/content/blogposts/blog-post-starter-pack.md).

To use the template, copy the above file and paste it in the appropriate year folder (i.e. `content/blogposts/2022/`). Rename the file to the slug the post will be, or something similar. Edit the markdown content accordingly, all available standardized components are ready to be changed right in this file!

## Adding images

Please use a `Figure` component to add images to your post. Optionally, you can include a `caption` to display a description below the image.

<!-- prettier-ignore -->
```html
<Figure
  src="https://storage.googleapis.com/sourcegraph-assets/blog/default_hero_social.png"
  alt="Sourcegraph thumbnail"
  caption="The Sourcegraph logo"
  link="https://https://sourcegraph.com/search.com"
/>

> ```

Reviewer: Instead of referencing all the components code here in the handbook, I think we should just reference the starter pack alone and maintain that as the source of truth. Then we can avoid maintaining two places at once and avoid outdated API mismatches. If we ever change the API on the about site, the blog starter pack will enforce us to update it since it will error/warn on build time. I would then make a note here in this guide to "please reference the blog starter pack." What do you think?

Reviewer 2: I see what you mean. So not show the syntax for HubSpot forms either? I can list all the components and point teammates to go to the Blog Post Starter Pack to see the syntax use.

Reviewer 2: Let me know thoughts on this phrasing! [92083e9](https://github.com/sourcegraph/handbook/pull/4186/commits/92083e927792184ac38b9b139a68ecb2aa741cd4)

Reviewer: Yeah I wouldn't show the code examples on the Handbook. I would also even move the frontmatter section to the blog starter pack and point people there too.

## Using the Blog Post Starter Pack

The Content Platform Team maintains a template Markdown file that exposes all of the content components available for use in the blog. You can directly copy and paste components from this file into your post. It's important to use the components provided for tables, images, alerts, youtube videos, external or self hosted videos, and HubSpot forms to ensure our blog posts are uniform in styling.

Access the template in the About repository: [`/content/blogposts/blog-post-starter-pack.md`](https://github.com/sourcegraph/about/tree/main/content/blogposts/blog-post-starter-pack.md).

To use the template, copy the above file and paste it in the appropriate year folder (i.e. `content/blogposts/2022/`). Rename the file to the slug the post will be, or something similar. Edit the markdown content accordingly, all available standardized components are ready to be changed right in this file!

<hr />

> To ensure uniformity between commonly used content items, please use the following components where appropriate. These are setup to accommodate screen responsiveness, accessibility concerns, and our brand spacing and theme designs.

Reviewer: ```suggestion
To ensure uniformity between commonly used content items, please use the following components where appropriate. These are setup to accommodate screen responsiveness, accessibility concerns, and meet our brand guidelines.
```

### Uploading images

- If you do not have a custom hero or social image, use this [default hero image](https://storage.googleapis.com/sourcegraph-assets/blog/default_hero_social.png).

> - Small images can be placed in the `/public/blog` directory and have the url of `/blog/example-image.jpg` in your markdown.

Reviewer: ```suggestion
- Small images can be placed in the `/public/blog` directory and have the url of `/blog/example-image.jpg` in your markdown. We also strongly encourage you to create a sub directory with the same slug as your post if you are including multiple files for your blog post. This keeps our assets organized per blog post. For example: `/blog/my-slug/image-name.png`, `/blog/my-slug/other-image-name.png`.
```

Fixing, editing, and updating a blog post on [about.sourcegraph.com](https://about.sourcegraph.com/blog/) is easy, can be done in minutes, and does not require running code locally.

This video shows the process from start to finish, although only those with repository push access will be able to squash and merge the change. Find a blogpost file in [About's repository](https://github.com/sourcegraph/about/tree/main/content/blogposts), the slug and file name are usually the same. Edit and preview the markdown in GitHub. Create a pull request from the browser. The change can be previewed again in staging, accessible from the `deploy/netlify` **Details** button.

<div className="container">
  <video loop autoplay muted playsinline>
    <source src="https://sourcegraphstatic.com/handbook/edit-blogpost-demo.webm" type="video/webm" />
    <source src="https://sourcegraphstatic.com/handbook/edit-blogpost-demo.mp4" type="video/mp4" />

>   </video>

Reviewer 2: Do you think this should be a youtube video?
